artcommon/artcommonlib/logutil.py

```diff
@@ -1,11 +1,82 @@
 import logging
+import re
 
 
 class EntityLoggingAdapter(logging.LoggerAdapter):
     def process(self, msg, kwargs):
         return '[%s] %s' % (self.extra['entity'], msg), kwargs
 
 
+class RedactingFilter(logging.Filter):
+    """ A filter to remove sensitive information like passwords, tokens, and keys from logs
+    https://docs.python.org/3/library/logging.html#filter-objects
+    """
+
+    PATTERNS = {
+        'requests_gssapi.gssapi_': [
+            (re.compile(r'Authorization header:(.+)', re.IGNORECASE), 'Authorization header: <redacted>'),
+        ]
+    }
+
+    def filter(self, record: logging.LogRecord):
+        if record.name in self.PATTERNS.keys():
+            self._redact(record)
+        return True
+
+    def _redact(self, record: logging.LogRecord):
+        msg = isinstance(record.msg, str) and record.msg or str(record.msg)
+        patterns = self.PATTERNS[record.name]
+        for pattern, replacement in patterns:
+            m = pattern.search(msg)
+            if m:
+                record.msg = pattern.sub(replacement, msg)
+                break
+
+
+def setup_logging(log_level: int, debug_log_path: str):
```

Comment:
This actually returns a value, so maybe it should be called `get_root_logger` or similar?

Reply:
I've removed the return value. The root logger is rarely used for printing logs. In case of the root logger is needed, we can always call `logging.getLogger()`.